Source/WebCore/crypto/gcrypt/CryptoKeyOKPGCrypt.cpp

```diff
@@ -91,24 +95,55 @@ std::optional<CryptoKeyPair> CryptoKeyOKP::platformGeneratePair(CryptoAlgorithmI
         return std::nullopt;
     }
 
-    // Retrieve the `q` data, which should be in the uncompressed point format for the X25519 keys.
-    // Validate the data size and the first byte (which should be 0x40).
-    size_t pubKeySize = namedCurve == NamedCurve::Ed25519 ? 32 : 33;
-    auto publicKeyData = mpiData(qMpi);
-    if (!publicKeyData || publicKeyData->size() != pubKeySize
-        || (namedCurve == NamedCurve::X25519
-            && !CryptoConstants::matches(publicKeyData->data(), 1, CryptoConstants::s_x25519UncompressedFormatLeadingByte))) {
+    auto q = mpiData(qMpi);
+    auto d = mpiData(dMpi);
+    if (UNLIKELY((!q || !d)))
```

Comment:
Nit: I think one pair of parentheses can be removed.

Reply:
true, thanks